script_library/cleanup-k8s.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+for line in $(helm ls -a | awk 'NR > 1 {print $1 }'); do
+    helm delete $line --purge;
+done
+
+for NS in openstack ceph nfs libvirt; do
+   helm ls --namespace $NS --short | xargs -r -L1 -P2 helm delete --purge
+done
+
+rm -rf /var/lib/openstack-helm/*
+rm -rf /var/lib/nova/*
+rm -rf /var/lib/libvirt/*
+rm -rf /etc/libvirt/qemu/*
+findmnt --raw | awk '/^\/var\/lib\/kubelet\/pods/ { print $1 }' | xargs -r -L1 -P16 sudo umount -f -l
+
+for pvc in $(kubectl get pvc -n openstack | awk ' NR > 1 { print $1 ; }'); do
+    kubectl delete pvc $pvc -n openstack;
+done
+
```

Comment:
I would add an additional loop to delete pv instances once the pvcs have been deleted. They will show a state of "failed", but they may actually still be consuming space in the rbd pool if they're not manually deleted.

Reply:
would
`kubectl delete pv -n openstack || true`

be enough?

Reply:
Yes, that should do it. I'm not sure if there are any states that would cause the delete command to ignore a pv, so it might be a good idea to add the '--all' flag as well.